src/Pos/Block/Slog/Logic.hs

```diff
@@ -303,8 +307,13 @@ slogRollbackBlocks blunds = do
         mconcat [forwardLinksBatch, inMainBatch]
 
 -- Common actions for rollback and apply.
-slogCommon :: MonadSlogApply ssc ctx m => LastBlkSlots -> m ()
+slogCommon
+    :: MonadSlogApply ssc ctx m
+    => LastBlkSlots
+    -> m ()
 slogCommon newLastSlots = do
     sanityCheckDB
     slogPutLastSlots newLastSlots
-    putSlottingData =<< GS.getSlottingData
+    -- TODO(ks): Again, we can use memory, and not break encapsulation.
```

Comment:
It's about `MonadSlotsData` and the fact we can use memory to fetch the data, which isn't the case on all these calls.